circuit_maintenance_parser/parsers/telstra.py

```diff
@@ -0,0 +1,78 @@
+"""Telstra parser."""
+import logging
+from typing import Dict, List
+
+import dateutil.parser as parser
+from bs4.element import ResultSet  # type: ignore
+
+from circuit_maintenance_parser.errors import ParsingError
+from circuit_maintenance_parser.parser import Html, Impact, CircuitImpact, Status
+
+# pylint: disable=too-many-branches
+
+
+logger = logging.getLogger(__name__)
+
+
+class ParserTelstra(Html):
+    """Notifications Parser for Telstra notifications."""
+
+    provider_type: str = "telstra"
+
+    # Default values for Telstra notifications
+    _default_provider = "telstra"
+    _default_organizer = "gpen@team.telstra.com"
+
+    def parse_html(self, soup, data_base):
+        """Execute parsing."""
+        data = data_base.copy()
+        try:
+            self.parse_tables(soup.find_all("table"), data)
+
+            return [data]
+
+        except Exception as exc:
+            raise ParsingError from exc
+
+    def parse_tables(self, tables: ResultSet, data: Dict):
+        """Parse Table tag."""
+        for table in tables:
+            for td_element in table.find_all("td"):
+                # TODO: We should find a more consistent way to parse the status of a maintenance note
+                if "Planned Maintenance has been scheduled" in td_element.text:
+                    data["status"] = Status("CONFIRMED")
+                elif "This is a reminder notification to notify that a planned maintenance" in td_element.text:
+                    data["status"] = Status("CONFIRMED")
+                else:
+                    continue
+                break
+            for th_element in table.find_all("th"):
+                if not th_element.string:
+                    continue
+                th_text = th_element.string.strip()
+                if th_text == "To:":
+                    data["account"] = th_element.next_sibling.next_sibling.string
```

Comment:
Might be nice to assign `th_element.next_sibling.next_sibling.string` to a variable.
```suggestion
                th_text = th_element.string.strip()
                td_text = th_element.next_sibling.next_sibling.string
                if th_text == "To:":
                    data["account"] = td_text
```

(etc.)

Additionally, would something like `th_element.next_element.string` be correct as well?

Reply:
Not sure what do you exactly mean with "Additionally, would something like th_element.next_element.string be correct as well?" 
but if you mean " td_text = th_element.next_element.string" this doesn't work because the HTML structure, and depending on the level it is the info I use next_sibling (same level) or next_element (whatever comes afterwards)

Reply:
Yeah, that's what I meant, sorry for the lack of clarity. Thanks for the explanation.